rust/connlib/clients/apple/build.rs

```diff
@@ -1,140 +1,147 @@
+    fn compile_swift(env: &Env) -> anyhow::Result<()> {
+        let swift_sdk = diva::Command::parse("xcrun --show-sdk-path --sdk")
+            .with_arg(env.sdk)
+            .run_and_wait_for_trimmed()?;
+        let swift_src_files = WalkDir::new(&env.swift_src_dir)
+            .into_iter()
+            .filter_map(Result::ok)
+            .filter_map(|entry| {
+                (entry.path().extension() == Some("swift".as_ref()))
+                    .then(|| entry.path().to_owned())
+            });
+        std::fs::create_dir_all(&env.swift_built_lib_dir)?;
+        diva::Command::parse("swiftc -emit-library -static")
+            .with_args(["-module-name", SWIFT_PKG_NAME])
+            .with_arg("-import-objc-header")
+            .with_arg(env.swift_src_dir.join(BRIDGING_HEADER))
+            .with_arg("-sdk")
+            .with_arg(swift_sdk)
+            .with_args(["-target", &env.triple])
+            // https://github.com/apple/swift-package-manager/blob/55006dce81ae70cd8f2b78479038423eeebde1e4/Documentation/Usage.md#setting-the-build-configuration
+            .with_parsed_args(if !env.release {
+                "-Onone -g -enable-testing"
+            } else {
+                "-O -whole-module-optimization"
+            })
+            .with_arg("-o")
+            .with_arg(env.swift_built_lib_dir.join(SWIFT_LIB_NAME))
+            .with_args(swift_src_files)
+            .with_cwd(&env.swift_pkg_dir)
+            .run_and_wait()?;
+        Ok(())
+    }
 
-fn link_swift(env: &Env) {
-    println!("cargo:rustc-link-lib=static={SWIFT_PKG_NAME}");
-    println!(
-        "cargo:rustc-link-search={}",
-        env.swift_built_lib_dir.display()
-    );
-    let xcode_path = diva::Command::parse("xcode-select --print-path")
-        .run_and_wait_for_trimmed()
-        .unwrap_or_else(|_| "/Applications/Xcode.app/Contents/Developer".to_owned());
-    println!("cargo:rustc-link-search={xcode_path}/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/");
-    println!("cargo:rustc-link-search=/usr/lib/swift");
-}
+    fn link_swift(env: &Env) {
+        println!("cargo:rustc-link-lib=static={SWIFT_PKG_NAME}");
+        println!(
+            "cargo:rustc-link-search={}",
+            env.swift_built_lib_dir.display()
+        );
+        let xcode_path = diva::Command::parse("xcode-select --print-path")
+            .run_and_wait_for_trimmed()
+            .unwrap_or_else(|_| "/Applications/Xcode.app/Contents/Developer".to_owned());
+        println!("cargo:rustc-link-search={xcode_path}/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/");
+        println!("cargo:rustc-link-search=/usr/lib/swift");
+    }
 
-fn main() -> anyhow::Result<()> {
     println!("cargo:rerun-if-env-changed={XCODE_CONFIGURATION_ENV}");
     let env = Env::gather();
     gen_bridges(&env);
     compile_swift(&env)?;
     link_swift(&env);
     Ok(())
 }
+
+#[cfg(not(any(target_os = "darwin", target_os = "ios", target_os = "ios-sim")))]
```

Comment:
Weird that Rust uses `darwin` in the target triple but `macos` for `target_os`
```suggestion
#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "ios-sim")))]
```